src/main/java/persistence/sql/ddl/dialect/database/Database.java

```diff
@@ -0,0 +1,36 @@
+package persistence.sql.ddl.dialect.database;
+
+import persistence.sql.ddl.dialect.Dialect;
+import persistence.sql.ddl.dialect.DialectResolution;
+import persistence.sql.ddl.dialect.exception.NotFoundDatabase;
+import persistence.sql.ddl.dialect.h2.H2Dialect;
+
+import java.util.Arrays;
+
+public enum Database {
+
+    H2 {
+        @Override
+        public Dialect createDialect() {
+            return new H2Dialect();
+        }
+
+        @Override
+        public boolean productNameMatchers(String databaseName) {
+            return "H2".equals(databaseName);
```

Comment:
```suggestion
            return this.name().equals(databaseName);
```
Enum 의 name 을 활용하면 String 을 제거해볼 수 있을것 같아요 👍